feast_component/component.py

```diff
@@ -23,34 +23,65 @@
 
 """
 
-from typing import Optional
+import os
+import pathlib
+import tempfile
+from typing import List, Optional
 
+import feast
+from google.protobuf.struct_pb2 import Struct
 from tfx.components.example_gen import component
-from tfx.components.example_gen import utils
 from tfx.dsl.components.base import executor_spec
-from tfx.examples.custom_components.presto_example_gen.presto_component import executor
 from tfx.proto import example_gen_pb2
 
-from protos import FeastConfig
+from . import executor
 
 
-class FeastExampleGen(component.QueryBasedExampleGen):    
+class FeastExampleGen(component.QueryBasedExampleGen):
 
     EXECUTOR_SPEC = executor_spec.BeamExecutorSpec(executor.Executor)
 
-    def __init__(self,
-                 conn_config: FeastConfig.FeastConnConfig,
-                 query: Optional[str] = None,
-                 input_config: Optional[example_gen_pb2.Input] = None,
-                 output_config: Optional[example_gen_pb2.Output] = None):
+    def __init__(
+        self,
+        repo_config: feast.RepoConfig,
+        entity_query: Optional[str] = None,
+        features: Optional[List[str]] = None,
```

Comment:
to make this consistent w feast APIs, maybe features should be Union[List[str], FeatureService] rather than two optional params. For instance, Feast's get_historical_features looks like (the feature_refs param is for backward compatibility that will get deprecated soon)

`def get_historical_features(
        self,
        entity_df: Union[pd.DataFrame, str],
        features: Optional[Union[List[str], FeatureService]] = None,
        feature_refs: Optional[List[str]] = None,
        full_feature_names: bool = False,
    ) -> RetrievalJob:`

Reply:
Updated to accept either list or a feaureservice object. Also accepting a string representation for featureservice as talked in today's meeting